gmso/core/parametric_potential.py

```diff
@@ -195,5 +151,14 @@ def from_template(cls, potential_template, parameters, topology=None):
                    expression=potential_template.expression,
                    independent_variables=potential_template.independent_variables,
                    parameters=parameters,
-                   topology=topology,
-                   template=False)
+                   topology=topology)
+
+    class Config:
+        fields = {
+            'topology_': 'topology',
+            'set_ref_': 'set_ref'
+        }
+        alias_to_fields = {
+            'topology': 'topology_'
```

Comment:
The set_ref variable is an internal constant and is exposed as readonly property. It may be provided during class initialization, however, cannot be changed using `__setattr__`